modules/scheduler/scheduler/src/main/java/org/motechproject/scheduler/service/impl/MotechSchedulerServiceImpl.java

```diff
@@ -786,12 +800,16 @@ private void validateJob(JobKey key) throws SchedulerException {
 
         JobDataMap map = detail.getJobDataMap();
 
-        if (map != null && !map.getBooleanValue(UI_DEFINED)) {
+        if (map != null && isJobUIDefined((Map<String, Object>) map.get(EVENT_METADATA))) {
             throw new MotechSchedulerException(String.format("Job is not ui defined:\n %s\n %s", key.getName(),
                     key.getGroup()));
         }
     }
 
+    private Boolean isJobUIDefined(Map<String, Object> jobMetadataMap) {
```

Comment:
please return primitive type here, additional this method should have jobdatamap parameter, all this conditions in if should be here to improve code readability